integration_test/common/common.go

```diff
@@ -43,7 +46,8 @@ type ExpectedMetric struct {
 	// Exactly one metric in each expected_metrics.yaml must
 	// have Representative set to true. This metric can be used
 	// to test that the integration is enabled.
-	Representative bool `yaml:"representative" validate:"excluded_with=Optional"`
+	Representative bool   `yaml:"representative" validate:"excluded_with=Optional"`
+	Platform       string `yaml:"platform,omitempty" validate:"omitempty,oneof=linux windows"`
```

Comment:
For agent metrics, we probably will use agent/uptime metrics as the representative metric.

This is indeed a very good question we might face in the future.

For agent metrics, there are some metrics that are Linux only or Windows only. `Platform` is used during testing(maybe later also be used public doc too) to see if this metric shall be tested on Linux or Windows or Both i think.

For 3rd party apps metrics, so far all 3rd party apps metrics are Linux only except IIS, MSSQL and active_directory_ds. But you are right on one thing, there could be one 3rd party app that are on both windows and linux, and when we support both platforms on such an integration, i can see below scenarios could happen:

- There are shared metrics <- great, then we could easily find one representative metrics for both platforms. (If platform is not specified then it's for both)
- Metrics are entirely mutual exclusive meaning for that integration, there ain't any shared metrics type by either platform. If so, that's the time we might see we will be in trouble. But it could possibly work with current setUp and just one metadata.yaml files.
```
expected_metrics:
- type: workflow.googleapis.com/abc.a
  value_type: INT64
  kind: CUMULATIVE
  monitored_resource: gce_instance
  platform: linux
  labels:
    device: .*
    direction: .*
- type: workflow.googleapis.com/abc.b
  value_type: INT64
  kind: CUMULATIVE
  monitored_resource: gce_instance
  platform: windows
  representative: true
  labels:
    device: .*
    direction: .*
- type: workflow.googleapis.com/abc.d
  value_type: INT64
  kind: CUMULATIVE
  platform: linux
  representative: true
  monitored_resource: gce_instance
  labels:
    device: .*
    direction: .*
```

We can make the validation check have such a check, if for all the metrics under expected_metrics, none of them are shared(having no platform specified meaning it's shared) and all has platform specified, for each platform, it shall have one and only one representative metrics.